File: frontend/src/types/Snapshot.ts
import type { SnapshotChange } from './SnapshotChange'
import { Bar } from '@/types/Bar'

export interface Snapshot {
  class: string
  symbol: string
  name: string
  exchange: string
  price: number
  prevClose: number
  prevCloseDate: string
  dailyHigh: number
  dailyLow: number
  dailyVolume: number
  change: SnapshotChange
  volumes: Array<{
    date: string,
    vol: number
  }>
  monthlyBars: Array<Bar>
  ytdChange: SnapshotChange
}

File: snapshots/snapshot.go
package snapshots

import (
	"github.com/phoobynet/market-deck-server/bars"
	"github.com/phoobynet/market-deck-server/helpers/numbers"
)

type Snapshot struct {
	Class         string                   `json:"class"`
	Symbol        string                   `json:"symbol"`
	Name          string                   `json:"name"`
	Exchange      string                   `json:"exchange"`
	Price         float64                  `json:"price"`
	PrevClose     float64                  `json:"prevClose"`
	PrevCloseDate string                   `json:"prevCloseDate"`
	DailyHigh     float64                  `json:"dailyHigh"`
	DailyLow      float64                  `json:"dailyLow"`
	DailyVolume   float64                  `json:"dailyVolume"`
	Change        numbers.NumberDiffResult `json:"change"`
	Volumes       []SnapshotVolume         `json:"volumes"`
	MonthlyBars   []bars.Bar               `json:"monthlyBars"`
	YtdBars       []bars.Bar               `json:"ytdBars"`
	YtdChange     numbers.NumberDiffResult `json:"ytdChange"`
}
